File: Project/Transfer/Move/ULM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MachShooting.Graphic;

namespace MachShooting
{
    /// <summary>
    /// 等速直線運動
    /// Uniform linear motion
    /// </summary>
    public class ULM : ITransfer
    {
        /// <summary>
        /// 対象のゲームオブジェクト
        /// </summary>
        private readonly GameObject go;

        /// <summary>
        /// 1Fの移動量
        /// </summary>
        private readonly Vec vec;

        /// <summary>
        /// 移動するF数
        /// </summary>
        private readonly int f;

        /// <summary>
        /// 現在のカウント
        /// </summary>
        private int count;

        public bool Need
        {
            get
            {
                return this.count < this.f;
            }
        }

        public ULM(GameObject go,Vec vec,int f)
        {
            this.go = go;
            this.vec = vec;
            this.f = f;
        }

        public void Draw()
        {
        }

        public List<AttackObject> Process()
        {
            if (this.Need)
            {
                this.go.X += this.vec.X;
                this.go.Y += this.vec.Y;

                this.count++;
            }
            return null;
        }
    }
}

File: Project/BuildData/script/command/ULM.lua
ULM=function(api,vec,f){
    local this={};

    local count=0;

    local vx,vy,f=0;

    this.isNeed=function()
        return count < f;
    end

    this.draw=function()
    end

    this.update=function()
        if count==0 then
            vx,vy=vec();
            f=f();
        end
        if this.isNeed
            api.X=api.X+vx;
            api.Y=api.Y+vy;

            count=count+1;
        end
    end

    return this;
}